lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -60,6 +61,202 @@ function dosomething_rogue_get_reportback_items($inputs)
 }
 
 /**
+  * Send a GET request to return all activiy matching a given
+  * query from Rogue.
+  *
+  * @param array $inputs
+  */
+function dosomething_rogue_get_activity($inputs)
+{
+  $client = dosomething_rogue_client();
+
+  $rogue_inputs = [
+    'filter' => [
+      'campaign_id' => dosomething_helpers_isset($inputs, 'campaigns'),
+      'campaign_run_id' => dosomething_helpers_isset($inputs, 'runs'),
+      'northstar_id' => dosomething_helpers_isset($inputs, 'user'),
+    ],
+    'limit' => dosomething_helpers_isset($inputs, 'count'),
+    'page' => dosomething_helpers_isset($inputs, 'page'),
+  ];
+
+  $response = $client->getActivity($rogue_inputs);
+
+  return $response;
+}
+
+/**
+  * Format a response from Rogue's GET /activity to match
+  * the response from Phoenix's GET /signups
+  *
+  * @param array $inputs
+  */
+function dosomething_rogue_format_activity($response)
+{
+  $data = [];
+
+  if ($response) {
+    $query = drupal_get_query_parameters();
+    $current_page = $query['page'];
+
+    // Set next page
+    if (is_null($current_page)) {
+      $query['page'] = 2;
+    } else {
+      $query['page'] = $query['page'] + 1;
+    }
+
+    $next_uri = url(current_path(), ['absolute' => TRUE, 'query' => $query]);
+
+    // Set previous page
+    if ($current_page < 2) {
+      $prev_uri = null;
+    } else {
+      $query['page'] = $current_page - 1;
+
+      $prev_uri = url(current_path(), ['absolute' => TRUE, 'query' => $query]);
+    }
+
+    // Include meta data
+    $data['meta'] = [
+      'pagination' => [
+        'total' => $response['meta']['pagination']['total'],
+        'per_page' => $response['meta']['pagination']['per_page'],
+        'current_page' => $response['meta']['pagination']['current_page'],
+        'total_pages' => $response['meta']['pagination']['total_pages'],
+        'links' => [
+          'prev_uri' => $prev_uri,
+          'next_uri'=> $next_uri,
+        ],
+      ],
+    ];
+
+    $data['data'] = [];
+
+    // Format each signup
+    foreach ($response['data'] as $signup) {
+      // Get campaign and run information for the signup data response
+        $northstar_user = dosomething_northstar_get_user($signup['northstar_id']);
+
+        // Get campaign information
+        $drupal_user = dosomething_user_get_user_by_northstar_id($signup['northstar_id']);
+
+        try {
+          $campaign_run = dosomething_helpers_get_current_campaign_run_for_user($signup['campaign_id'], $drupal_user); // @TODO - get errors when calling this sometimes it seems (500 - no campaign data found??????)
+        } catch(Exception $e) {
+          // the regular Phoenix endpoint doesn't seem to handle this so should we just leave it? unclear
```

Comment:
Yea, I'm pretty sure the phoenix endpoint doesn't support runs. let's leave it unhandled.